framework/auth/campaigns.py

```diff
@@ -1,68 +1,98 @@
-import httplib as http
-
 import furl
+
 from werkzeug.datastructures import ImmutableDict
-from framework.exceptions import HTTPError
 
 from website import mails
 from website.settings import DOMAIN
 
-CAMPAIGNS = ImmutableDict({
+CAMPAIGNS = {
     'prereg': {
         'system_tag': 'prereg_challenge_campaign',
-        'redirect_url': lambda: furl.furl(DOMAIN).add(path='prereg/').url,
+        'redirect_url': furl.furl(DOMAIN).add(path='prereg/').url,
         'confirmation_email_template': mails.CONFIRM_EMAIL_PREREG,
     },
     'institution': {
         'system_tag': 'institution_campaign',
-        'redirect_url': lambda: ''
+        'redirect_url': ''
     },
     'erpc': {
         'system_tag': 'erp_challenge_campaign',
-        'redirect_url': lambda: furl.furl(DOMAIN).add(path='erpc/').url,
+        'redirect_url': furl.furl(DOMAIN).add(path='erpc/').url,
         'confirmation_email_template': mails.CONFIRM_EMAIL_ERPC,
     },
     # Various preprint services
     # Each preprint service will offer their own campaign with appropriate distinct branding
     'osf-preprints': {
         'system_tag': 'osf_preprints',
-        'redirect_url': lambda: furl.furl(DOMAIN).add(path='preprints/').url,
-        'confirmation_email_template': mails.CONFIRM_EMAIL_PREPRINTS_OSF,
+        'redirect_url': furl.furl(DOMAIN).add(path='preprints/').url,
+        'confirmation_email_template': mails.CONFIRM_EMAIL_PREPRINTS('osf', 'OSF'),
+        'proxy_login': True,
+        'branded': False,
+        'provider': 'OSF'
+    },
+}
+
+providers = [
```

Comment:
Agreed. This should be classes and database objects, given we may have more campaigns and branded services coming up. I am not sure if we should make the change in this PR. I prefer having another dedicated PR. An alternative is to let Django-OSF take care of this since it is already refactoring models.